src/redisearch_rs/rqe_iterators/src/inverted_index_it.rs

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+//! Inverted index iterator implementation
+use ffi::t_docId;
+use inverted_index::{DecodedBy, IndexReader, RSIndexResult};
+
+use crate::{RQEIterator, RQEIteratorError, SkipToOutcome};
+
+/// An iterator over numeric inverted index entries.
+///
+/// This iterator provides full index scan to all document IDs in a numeric inverted index.
+/// It is not suitable for queries.
+///
+/// # Type Parameters
+///
+/// * `'index` - The lifetime of the index being iterated over.
+/// * `E` - The encoder used to encode the inverted index.
+pub struct NumericFull<'index, E: DecodedBy> {
```

Comment:
Since this is for numeric inverted indices, is the generic `E` needed? Why not set it to use the `Numeric` decoder directly?

Reply:
You're totally right, I over generalized for no reason. Fixed.